src/main/java/lotto/model/WinningResults.java

```diff
@@ -1,25 +1,80 @@
 package lotto.model;
 
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 
 public class WinningResults {
+    private final WinningLotto winningLotto;
+    private final Lottos lottos;
+    private final List<WinningResult> results;
 
-    private final List<WinningResult> responses;
+    public WinningResults(WinningLotto winningLotto, Lottos lottos) {
+        this.winningLotto = winningLotto;
+        this.lottos = lottos;
+        Map<Rank, Integer> ranks = scoreRanks();
+        this.results = toResults(ranks);
+    }
+
+    private Map<Rank, Integer> scoreRanks() {
+        Map<Rank, Integer> ranks = new LinkedHashMap<>();
+        initRanks(ranks);
+        saveRanks(ranks);
```

Comment:
메서드에서 매개변수 ranks를 인자로 넣어주고 이를 직접 변경하고 있군요. 이는 ranks가 어디서 변경됐는지 파악하기 쉽지 않으며 디버깅을 어렵게 만들 수 있습니다.
또 initRanks가 단순히 Rank를 키로 가지고 있는 Map을 초기화하는 역할을 하고 있는데요, saveRanks 메서드 에서 호출하는 것도 괜찮아 보입니다.
```suggestion
        Map<Rank, Integer> savedRanks = initRanks(ranks);
```
함수들의 return값으로 변경된 ranks를 반환하게 명시적으로 표현해보는건 어떨까요?

Reply:
좋은 의견인 것 같습니다!
수정해볼게요!